projects/components/src/datagrid/directives/select-directive.ts

```diff
@@ -0,0 +1,46 @@
+/*!
+ * Copyright 2019 VMware, Inc.
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+import { OnChanges } from '@angular/core';
+import { EventEmitter } from '@angular/core';
+import { Directive, Input } from '@angular/core';
+import { ClrDatagrid } from '@clr/angular';
+import { Host } from '@angular/core';
+import { GridSelectionType } from '../datagrid.component';
+
+@Directive({
+    selector: '[vcdComponentGridSelection]',
+})
+export class ComponentGridSelectionDirective<R> implements OnChanges {
+    @Input() gridSelectionType: GridSelectionType;
```

Comment:
All Inputs and outputs should have documentation. Why was this directive needed, couldn't we just have a [ViewRef](https://angular.io/api/core/ViewChild) to the grid and do the work from vcd-grid?